File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public long KthLargestLevelSum(TreeNode root, int k) {
        PriorityQueue<long, long> pq = new();
        Queue<TreeNode> queue = [];
        queue.Enqueue(root);

        while(queue.Count > 0)
        {
            int levelSize = queue.Count;
            long sum = 0;

            for(int i = 0; i < levelSize; i++)
            {
                var node = queue.Dequeue();

                sum += node.val;

                if(node.left!=null)
                    queue.Enqueue(node.left);
                if(node.right!=null)
                    queue.Enqueue(node.right);
            }
            pq.Enqueue(sum, -sum); // Negative priority for descending sorting
        }    

        // Number of levels lesser than k
        if(k > pq.Count)
            return -1;

        // Pop k-1 times to reach the kth sum
        while(k > 1){
            pq.Dequeue();
            k--;
        }

        return pq.Peek();
    }
}

File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthLargestLevelSum = function (root, k) {
    let q = [root];
    let sums = [];
    while (q.length) {
        let newQ = [];
        let sum = 0;
        for (let i = 0; i < q.length; i++) {
            const node = q[i];
            sum = sum + node.val;
            if (node.left) newQ.push(node.left);
            if (node.right) newQ.push(node.right);
        }
        sums.push(sum);
        q = newQ;
    }

    sums.sort((a, b) => b - a);
    
    return sums[k - 1] || -1;
};